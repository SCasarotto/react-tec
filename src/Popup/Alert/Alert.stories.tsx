import React from 'react';

import { Story, Meta } from '@storybook/react';

import { Alert, AlertProps } from './Alert';

export default {
  title: 'Popup/Alert',
  component: Alert,
  argTypes: {},
} as Meta;

const Template: Story<AlertProps> = (args) => <Alert {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'Alert!',
  message: 'You have been alerted to stay alert.',
  visible: true,
  children: (
    <p style={{ textAlign: 'center' }}>
      Some extra components passed into the body
    </p>
  ),
};
