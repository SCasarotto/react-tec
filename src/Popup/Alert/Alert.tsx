import React from 'react';

import { PopupProps } from '../Popup';
import { Popup, Title, Message, Button } from './styledComponents';

export interface AlertProps extends PopupProps {
  title?: string;
  message?: string;
  onClick?(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
  buttonTitle?: string;
}
/**
 *
 * Alert is a simple alert popup typically used to display error or success messages.
 *
 */
export const Alert: React.FC<AlertProps> = (props) => {
  const {
    title,
    message,
    onClick,
    buttonTitle = 'Okay',
    visible,
    className = '',
    children,
  } = props;

  return (
    <Popup visible={visible} className={`tec-alert ${className}`}>
      {title && <Title className='tec-alertTitle'>{title}</Title>}
      {message && <Message className='tec-alertMessage'>{message}</Message>}
      {children}
      <Button onClick={onClick} className='tec-alertButton'>
        {buttonTitle}
      </Button>
    </Popup>
  );
};
