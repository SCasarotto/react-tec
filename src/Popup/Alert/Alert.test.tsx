import React from 'react';

import { render } from '@testing-library/react';

import { Alert } from './Alert';

describe('Alert', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(<Alert visible />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(<Alert visible className='myClassName' />);
    expect(container.firstChild).toHaveClass('myClassName');
  });
});
