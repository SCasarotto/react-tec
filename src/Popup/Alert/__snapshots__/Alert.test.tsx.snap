// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches snapshot 1`] = `
<DocumentFragment>
  .c3 {
  position: relative;
  display: block;
  width: 100%;
  font-size: 18px;
  padding: 5px 20px;
  text-align: center;
  cursor: pointer;
  border-radius: 5px;
  box-sizing: border-box;
  margin: 0px;
  border: 1px solid;
  -webkit-transition: color 0.2s ease-in,background-color 0.2s ease-in, border 0.2s ease-in;
  transition: color 0.2s ease-in,background-color 0.2s ease-in, border 0.2s ease-in;
}

.c3:hover,
.c3:active {
  border: 1px solid;
}

.c3:disabled {
  cursor: not-allowed;
  border: 1px solid;
}

.c0 {
  position: fixed;
  width: 100%;
  height: 100%;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.5);
  z-index: 1000;
  overflow-y: auto;
  -webkit-transition: visibility 0.25s ease-in-out,opacity 0.25s ease-in-out;
  transition: visibility 0.25s ease-in-out,opacity 0.25s ease-in-out;
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: relative;
  border-radius: 20px;
  margin: 50px auto;
  max-width: 650px;
}

.c1 .PopupContent {
  width: 90%;
  max-width: 600px;
  margin: 100px auto;
}

.c4 {
  border-top: 1px solid;
  border-right: none;
  border-bottom: none;
  border-left: none;
  line-height: 2;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-left-radius: 20px;
  border-bottom-right-radius: 20px;
  -webkit-transition: border 0.2s ease-in,color 0.2s ease-in, background-color 0.2s ease-in;
  transition: border 0.2s ease-in,color 0.2s ease-in, background-color 0.2s ease-in;
}

.c4:active,
.c4:hover {
  border-top: 1px solid;
  border-right: none;
  border-bottom: none;
  border-left: none;
}

<div
    class="c0 tec-popup c1 tec-alert "
  >
    <div
      class="c2 tec-popupContent"
    >
      <button
        class="c3 tec-button c4 tec-alertButton"
        type="button"
      >
        Okay
      </button>
    </div>
  </div>
</DocumentFragment>
`;
